metrics-exporter-prometheus/src/exporter/http_listener.rs

```diff
@@ -59,11 +59,11 @@ impl HttpListeningExporter {
                     continue;
                 }
             };
-            self.process_tcp_stream(stream).await;
+            self.process_tcp_stream(stream);
         }
     }
 
-    async fn process_tcp_stream(&self, stream: TcpStream) {
+    fn process_tcp_stream(&self, stream: TcpStream) {
```

Comment:
This change is because the method doesn't actually await at all. I'm assuming that this is the right change, based on the usual case that a loop that handles incoming TCP requests usually wants to spawn a task and continue waiting for the next incoming requests. The other thing that could be meaningful here would be to wait for the request to be handled (e.g. await the task::spawn call). I think it's more likely the first, but could be wrong.

Reply:
I'm not sure if you're saying you want to allow the lint here, or that the change is fine.